jf_agent/data_manifests/git/generator.py

```diff
@@ -18,6 +18,13 @@ class UnsupportedGitProvider(Exception):
     pass
 
 
+def get_instance_slug(git_config, is_multi_git_config: bool, endpoint_git_instances_info) -> str:
```

Comment:
I don't like the pattern that this function has to be parameterized with `is_multi_git_config`. Could we instead have this function inspect the config and determine that itself?

Reply:
I view it as leading to later complexity and bugs to have part of the necessary functionality live in this function and part not.

Reply:
@nathanieltalbot I don't know if it's an indicator that it's specifically a mulit-git config, but it is impossible to have a multi git config WITHOUT a slug for each: https://github.com/Jellyfish-AI/jf_agent/blob/3f37c6025ecd34ffc78026504dd2eb5b8e25083d/jf_agent/config_file_reader.py#L386

That logic you linked in the above comment should hold, because a single git instance MIGHT have a slug provided, but if it doesn't we can fall back on the API. If we're in a multi config, than we know it WILL have a slug

Reply:
Agree with delany here, it'd be nice to decouple the `is_multi_git_config` logic here. Also it would be nice to add some type casting: `(git_config: GitConfig, is_multi_git_config: bool, endpoint_git_instances_info: dict)`